src/main/java/org/elasticsearch/action/admin/cluster/repositories/verify/TransportVerifyRepositoryAction.java

```diff
@@ -69,7 +69,7 @@ protected VerifyRepositoryResponse newResponse() {
 
     @Override
     protected ClusterBlockException checkBlock(VerifyRepositoryRequest request, ClusterState state) {
-        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA, "");
+        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA_WRITE, "");
```

Comment:
Oh, I forgot to mention that the Put Repository operation (which has METADATA_WRITE) executes by default the verification operation once the repository is registered so it seems more coherent to me. Anyway I'll be glad to have Igor opinion :)